### DISCUSSION TITLE
Adding `@discardableResult` to `.withExhaustivity()` ?

### DISCUSSION THREAD

**Discussion by LudvigIversen** [pointfreeco/swift-composable-architecture#2419] [Ideas]
Hello!

When working with test there are scenarios were we have a `.send()` operation followed by a `.receive()` like this for example 

```swift 
func test_() {
        let store = TestStore(
            initialState: Feature.State(),
            reducer: Feature()
        )

        store.send(.actionA)

        store.receive(.actionB)
    }
```

Now, let us say that `actionA` does some state changes that we for any reason don't want to assert on, in my case it is because the changes are outside of the scope of our test. 
Now, there is functionality to do this using the `withExhaustivity()` function, like so;

```swift 
func test_() {
        let store = TestStore(
            initialState: Feature.State(),
            reducer: Feature()
        )

       store.withExhaustivity(.off(showSkippedAssertions: true)) {
            store.send(.actionA)
        }
  
        store.receive(.actionB)
    }
```

Now this works, but there is an annoyance with this, and that is the warning that appears notifying us that the return of `withExhaustivity` is unused. To avoid this you can of course just put `_ =` in front of the call, however I believe that this is something that should be avoided.  

It is possible to achieve this by annotating the function with `@discardableResult` which would result in not needing the `_ =` whilst also avoiding the warning.

Understandable if there are reasons as to why this is not already the case, but if so, I would gladly hear that reason. Otherwise I believe this could be a great quality of life change.

**Comment by mbrandonw** [comment]
Hi @LudvigIversen, ideally `@discardableResult` would propagate up so that if you return a discardable result from a function that is not marked as such it will be discardable. We [opened an issue](https://github.com/apple/swift/issues/57477) about this long ago, but sadly it's not likely to ever be fixed.

And also we don't think it's appropriate to discard the result of `withExhaustivity` because if you do mean to return something from it, it will be silently ignored. It's worth noting that many of Swift/SwiftUI's tools have this same problem. For example, this:

```swift
withAnimation(.default) {
  somethingThatDiscardsResult()
}
```

…will also have a warning.

As such we don't think there's anything we can really do about it, and sadly you just have to litter `_ = ` throughout your code :/

**Reply by LudvigIversen** [reply]
Hello @mbrandonw , thank you for answering and explaining your reasoning!
